### DISCUSSION TITLE
my textual program not appear normally in powershell or cmd of the window

### DISCUSSION THREAD

**Discussion by MohamedElgamal** [Textualize/textual#3448] [Q&A] [answered]
Hello everyone,
Tried to run the program on the powershell of windows but it is not appeared correctly like linux
![image](https://github.com/Textualize/textual/assets/25607813/141545d5-22cd-458b-8ac2-15f83e9cbe5d)
also the program will be accessed using SSH by xterm terminal is the problem will still appear ?

**Comment by TomJGooding** [comment, marked as answer]
Try installing [Windows Terminal](https://apps.microsoft.com/store/detail/windows-terminal/9N0DX20HK7010), or changing your terminal font.

Windows Terminal seems to have font-fallback, unlike the classic Windows conhost where missing glyphs are replaced with question marks.

See this discussion for more details:

- #3117

**Reply by MohamedElgamal** [reply]
Thanks, the problem is fixed using , DevaVuSansM Nerd Font but also when i used bitvise terminal and setting the terminal protocol to xterm, the color of the program not appearing correctly it seems it has low range of colors as follow
![image](https://github.com/Textualize/textual/assets/25607813/d4520107-fb4e-4986-94a3-9bddc510f0d6)

i can't move forward by using window terminal because our base users use bitvise

**Reply by TomJGooding** [reply]
Could you run `textual diagnose` from your Bitvise terminal and paste the results?

The textual command is provided by [textual-dev](https://textual.textualize.io/guide/devtools/).

**Reply by MohamedElgamal** [reply]
unfortunately, can't run the textual-dev on my device because installing packages of python on my device is managed by sys admin,
but i am working on terminal protocol xterm not xterm 256 which has lower range of color, if there is any module in textual to handle the colors appearance in lower terminal color range?
sorry for my late response to you.

**Reply by TomJGooding** [reply]
So you think the Bitvise terminal emulator is really limited to only 8/16 colors? What does running `echo $TERM`  and  ` tput colors` show? Sorry I'm not sure I understand your question about a Textual module.

**Reply by MohamedElgamal** [reply]
yes the bitvise terminal is limited to 16 colors because the terminal options is set to xterm
here is the result of commands 
![image](https://github.com/Textualize/textual/assets/25607813/3cb15af6-b629-4cee-ad1b-89f047c98d15)
